zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java

```diff
@@ -327,4 +319,55 @@ public static ResponseEntity<Object> mapDocumentHandlingExceptionToResponse(
       final DocumentException e) {
     return mapProblemToResponse(mapDocumentHandlingExceptionToProblem(e));
   }
+
+  public static ProblemDetail mapCamundaSearchExceptionToProblem(CamundaSearchException cse) {
+    final CamundaSearchException.Reason reason =
+        (cse.getReason() == null) ? CamundaSearchException.Reason.UNKNOWN : cse.getReason();
+    final Throwable cause = (cse.getCause() == null) ? cse : cse.getCause();
+    final String title =
+        (reason == CamundaSearchException.Reason.UNKNOWN)
+            ? cause.getClass().getName()
+            : reason.name();
+    final String errorMessage = cause.getMessage();
```

Comment:
🔧 I think we can simplify the prolog of this method a bit.
* The `CamundaSearchException` should always have a reason. Let's enforce this in the exception itself by setting to UNKNOWN if `null` was provided. We can remove the check here.
* The reason should always be the `title`.
* The exception message should be what the search exception brings along. We can add information if needed in the switch cases.

```suggestion
    final CamundaSearchException.Reason reason = cse.getReason();
    final String title = reason.name();
    final String errorMessage = cse.getMessage();
```